torch/csrc/distributed/c10d/ProcessGroupWrapper.cpp

```diff
@@ -62,18 +72,56 @@ struct CollectiveFingerPrint {
     verify_tensors(inp, pg);
   }
 
+  // Takes a serialized fingerprint from
+  // CollectiveFingerPrint::serialize_fingerprint and deserializes it back to a
+  // CollectiveFingerPrint struct
+  CollectiveFingerPrint deserialize_fingerprint(at::Tensor serialized_tensor) {
```

Comment:
Thanks Rodrigo, I agree with you the manual deserialization is not ideal.

I think the challenge of using a library like flatbuffers / protobuf is that it also requires a change in how we want the nodes to communicate. Since we right now using `allgather` to transfer information between process group wrappers, then the data needs to be in a tensor form. If we use a 3rd party library we would either need to also to somehow transform the serialized data into tensor to continue using `allgather` or communicate the serialized data separately somehow.